letsencrypt/Dockerfile

```diff
@@ -1,15 +1,33 @@
 ARG BUILD_FROM
 FROM $BUILD_FROM
 
-# Setup base
+# setup base
 ARG CERTBOT_VERSION
-RUN apk add --no-cache \
-        openssl libffi musl libstdc++ \
+
+RUN apk add --no-cache --update \
+        openssl libffi musl libstdc++ wget tar python python-dev py-pip gcc \
     && apk add --no-cache --virtual .build-dependencies \
-        g++ musl-dev openssl-dev libffi-dev \
+    	g++ musl-dev openssl-dev libffi-dev \
     && pip3 install --no-cache-dir certbot==${CERTBOT_VERSION} \
     && apk del .build-dependencies
 
+# Install the DNS plugin
+RUN pip3 install certbot-dns-cloudflare==${CERTBOT_VERSION} \
```

Comment:
You don't need a new layer. You can move that into same command as you install cerbot.